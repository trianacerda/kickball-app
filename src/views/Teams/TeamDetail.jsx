import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeamById } from '../../services/teams';

function TeamDetail({ label, match }) {
  const { teamId } = match.params;
  const [team, setTeam] = useState(null);
  const [loading, setLoading] = useState(true);

  return (
    <>
      <h1>{label}</h1>
      <p>
        <Link to="/teams" className="team-link">
          Back to Team Page
        </Link>
      </p>
      <h1>{team.name}</h1>
      <p>
        From {team.city}, {team.state}
      </p>
      <ul>
        {team.players.map((player) => {
          return (
            <li key={player.id}>
              {player.position} - {player.name}
            </li>
          );
        })}
      </ul>
    </>
  );
}

export default TeamDetail;
