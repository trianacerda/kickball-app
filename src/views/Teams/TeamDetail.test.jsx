import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import TeamDetail from './TeamDetail';

it('should render a detailed view of an individual team by id', async () => {
  const { container } = render(
    <MemoryRouter>
      <TeamDetail match={{ params: { teamId: '1' } }} />
    </MemoryRouter>
  );

  screen.getByText('Please wait, the team is loading...');
  const teamName = await screen.findByText('Tala Loves Snacks', { name: 'team-name' });

  expect(teamName).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
