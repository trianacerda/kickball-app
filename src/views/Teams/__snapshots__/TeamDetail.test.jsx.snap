// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a detailed view of an individual team by id 1`] = `
<div>
  <p>
    <a
      class="home-link"
      href="/"
    >
      Home
    </a>
  </p>
  <p>
    <a
      class="team-link"
      href="/teams"
    >
      Teams
    </a>
  </p>
  - 
  <h1
    name="team-name"
  >
    Tala Loves Snacks
  </h1>
  <p>
    From 
    Sensitive
    , 
    OR
  </p>
  <ul>
    <li>
      Position: 
      Catcher
       - Name:
      Jonny  
Clutch
    </li>
    <li>
      Position: 
      First Base
       - Name:
      Charleston Dax
    </li>
    <li>
      Position: 
      Second Base
       - Name:
      Crispin Cruz
    </li>
    <li>
      Position: 
      Pitcher
       - Name:
      Declan Ebenezer
    </li>
    <li>
      Position: 
      Third Base
       - Name:
      Ike Jaspin
    </li>
    <li>
      Position: 
      Left Field 
       - Name:
      Kai Judson
    </li>
    <li>
      Position: 
      Middle Field
       - Name:
      Kayden Kale
    </li>
    <li>
      Position: 
      Right Field
       - Name:
      Slash Taden
    </li>
    <li>
      Position: 
      Big Kicker
       - Name:
      Tevin Zee
    </li>
  </ul>
</div>
`;
